specs/core/0_beacon-chain.md

```diff
@@ -663,13 +662,13 @@ full_deposit_count: uint256
 @payable
 @public
 def deposit(deposit_input: bytes[2048]):
-    assert msg.value >= as_wei_value(MIN_DEPOSIT, "ether")
-    assert msg.value <= as_wei_value(MAX_DEPOSIT, "ether")
+    assert msg.value >= MIN_DEPOSIT_AMOUNT
+    assert msg.value <= MAX_DEPOSIT_AMOUNT
 
     index: uint256 = self.deposit_count + TWO_TO_POWER_OF_TREE_DEPTH
-    msg_gwei_bytes8: bytes[8] = slice(concat("", convert(msg.value / GWEI_PER_ETH, bytes32)), start=24, len=8)
-    timestamp_bytes8: bytes[8] = slice(concat("", convert(block.timestamp, bytes32)), start=24, len=8)
-    deposit_data: bytes[2064] = concat(msg_gwei_bytes8, timestamp_bytes8, deposit_input)
+    amount: bytes[8] = slice(concat("", convert(msg.value / GWEI_PER_ETH, bytes32)), start=24, len=8)
```

Comment:
I'm tempted to abstract away in a helper function the (somewhat horrendous) hack of converting `x` to  `slice(concat("", convert(x, bytes32)), start=24, len=8)`. It's used 3 times in the Vyper contract.

Reply:
lol

Will come at a cost in gas. I'm not opposed unless we end up needing to shave down gas costs.

Reply:
That sounds good. Let me know if you want the abstraction to happen in this PR